src/main/java/com/google/sps/servlets/SaveQuestionsFromBankServlet.java

```diff
@@ -0,0 +1,126 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.DatastoreFailureException;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.flogger.FluentLogger;
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet that saves selected questions to the test.
+* @author Klaudia Obieglo
+*/
+@WebServlet("/saveQuestionsFromBank")
+public class SaveQuestionsFromBankServlet extends HttpServlet {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  @Override
+  public void doPost(final HttpServletRequest request,
+      final HttpServletResponse response) throws IOException {
+    /*Saves the selected questions from the question bank to the test
+    * that the user wants the questions saved to
+    */
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      logger.atWarning().log("User is not logged in.");
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED,
+          "You are not authorised to view this page");
+      return;
+    }
+    logger.atInfo().log("user=%s is logged in", userService.getCurrentUser());
+    String ownerID = userService.getCurrentUser().getEmail();
+
+    String testName = request.getParameter("test");
+    String[] questionsList = request.getParameterValues("question");
+    if (testName == null || questionsList == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      logger.atWarning().log("One or more null parameters");
+      return;
+    }
+    try {
+      for (int i = 0; i < questionsList.length; i++) {
+        addQuestionToExamList(Long.valueOf(questionsList[i]), ownerID, testName);
+        response.getWriter().println("Successfully added Question "
```

Comment:
How do you know that the question was successfully added? addQuestionToExamList(...) returns void and catches all the exceptions internally.